web-client/public/js/graph.js

```diff
@@ -93,74 +93,98 @@ var drawGraph = function (nodes, links, positiveWeights, negativeWeights, sheetT
+        if (!adaptive) {
+            $container.removeClass(CURSOR_CLASSES);
+        }
+    };
 
-    var drag = force.drag()
-      .origin(function (d) {
-          return d;
-      })
-      .on("dragstart", dragstart);
+    var zoomDrag = d3.drag()
+        .on("start", zoomDragStarted)
+        .on("drag", zoomDragged)
+        .on("end", zoomDragEnded);
 
     var manualZoom = false;
-    var zoom = d3.behavior.zoom()
-    .center([width / 2, height / 2])
-    .scaleExtent([minimumScale, maximumScale])
-    .on("zoom", zoomed);
 
     var svg = d3.select($container[0]).append("svg")
         .attr("width", width)
-        .attr("height", height)
-        .call(zoom).on("wheel.zoom", null) // disables mouse wheel zooming
-      .append("g") // required for zoom to work
+        .attr("height", height);
+
+    var zoomContainer = svg.append("g") // required for zoom to work
         .attr("class", "boundingBox")
         .attr("width", width)
-        .attr("height", height)
-        .append("g"); // appended another g here...
+        .attr("height", height);
 
-    d3.select("svg").on("dblclick.zoom", null); // disables double click zooming
+    var boundingBoxContainer = zoomContainer.append("g"); // appended another g here...
 
-  // This rectangle catches all of the mousewheel and pan events, without letting
-  // them bubble up to the body.
-    var innerRect = svg.append("rect") // eslint-disable-line no-unused-vars
-                     .attr("width", width)
-                     .attr("height", height)
-                     .style("fill", "none")
-                     .style("pointer-events", "all")
-                     .attr("stroke", adaptive ? "none" : "#9A9A9A")
-                     .append("g");
+    var zoom = d3.zoom()
+      .scaleExtent([1 / 2, 4])
+      .on("zoom", zoomed);
 
+    svg.style("pointer-events", "all").call(zoomDrag);
 
     function zoomed () {
-      // this part is not working
-        if (!adaptive) { // Limit to viewport
-            var scale = zoom.scale();
-            var scaledWidth = scale * width;
-            var scaledHeight = scale * height;
-            var maxX = width - scaledWidth;
-            var maxY = height - scaledHeight;
-
-            d3.event.translate[0] = Math.min(Math.max(d3.event.translate[0], 0), maxX);
-            d3.event.translate[1] = Math.min(Math.max(d3.event.translate[1], 0), maxY);
-            zoom.translate([d3.event.translate[0], d3.event.translate[1]]);
-        }
-        svg.attr("transform", "translate(" + zoom.translate() + ")scale(" + d3.event.scale + ")");
-
+        zoomContainer.attr("transform", d3.event.transform);
     }
 
+    d3.select("svg").on("dblclick.zoom", null); // disables double click zooming
+
+  // This rectangle catches all of the mousewheel and pan events, without letting
+  // them bubble up to the body.
+    var innerRect = boundingBoxContainer.append("rect") // eslint-disable-line no-unused-vars
```

Comment:
Why is `no-unused-vars` disabled here? Is `innerRect` never used? If so, then ditch it—I don't see why the `innerRect` declaration needs to be kept.

Reply:
Yes `innerRect` is not used. We got rid of the declaration and tested it-- works fine.